pkg/events/newentry/new_entry.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package newentry
+
+import (
+	"strings"
+	"time"
+
+	"github.com/sigstore/rekor/pkg/events"
+	"golang.org/x/exp/slices"
+
+	rekor_pb "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1"
+)
+
+const (
+	Name   = "dev.sigstore.rekor.events.v1.NewEntry"
+	Source = "/createLogEntry"
+)
+
+var ty *events.EventType
+
+func init() {
+	empty := &rekor_pb.TransparencyLogEntry{}
+	ty = events.RegisterType(Name, Source, empty.ProtoReflect().Descriptor())
+}
+
+func New(id string, entry *rekor_pb.TransparencyLogEntry, subjects []string) (*events.Event, error) {
+	slices.Sort(subjects) // Must be sorted for consistency.
+	attrs := map[string]any{
+		"time":                   time.Unix(entry.GetIntegratedTime(), 0),
+		"rekor_entry_kind":       entry.GetKindVersion().GetKind(),
```

Comment:
We'd also need version, since kind+version tells the caller how to parse the entry.

Reply:
I wonder if we should remove entry kind, to discourage filtering only by a kind that the searcher is aware of. For example, if an attacker steals credentials and knows that someone is filtering by a specific kind, they could upload a signed package for a different kind.

Reply:
With GCP PubSub, the cost savings can happen on both ends. Messages filtered server-sided are not sent to the client, and thus the consumer is not billed for the PubSub throughput (per byte again). The intent is to allow consumers to save cost by ignoring events they don't care about. I agree this could bite people, so the docs should have a large warning regarding this.

Reply:
I'm fine with keeping this with appropriate documentation.